lib/relation-mappings/many-to-many.ts

```diff
@@ -43,8 +43,12 @@ export default function generateManyToManyRelationMapping(
     joinTable = `${ ObjectionModel.tableName }_${ RelatedObjectionModel.tableName }`;
   }
 
-  mapping.join.through.from = `${ joinTable }.${ camelCase(type) }Id`; // i.e. from: 'Post_Tag.postId'
-  mapping.join.through.to = `${ joinTable }.${ camelCase(relatedType) }Id`; // i.e. from: 'Post_Tag.tagId'
+  let foreignKeyForRelationship = options.foreignKeyForRelationship && typeof options.foreignKeyForRelationship === 'function' ?
```

Comment:
Given my above comments about passing the descriptor rather than the model into this method, perhaps we can move this logic in the method itself as well? Something like `options.foreignKey || <camel case logic here>`.

Also, I don't see a use case for having `options.foreignKey` be a function - do you have an example of where a hardcoded string would not suffice?

Reply:
Yeah, I can't think of a specific case where it would have to be a function.